src/features/task-list/taskListSlice.ts

```diff
@@ -36,8 +40,76 @@ export const taskListSlice = createSlice({
         setTasksStatus: (state, action) => {
             state.taskStatus = action.payload;
         },
+
+        addNewTask: (state, action) => {
+            if (action.payload?.status?.toLowerCase() === 'todo') {
+                state.taskStatus.todo = (
+                    parseInt(state.taskStatus?.todo as string) + 1
+                ).toString();
+            } else if (action.payload?.status?.toLowerCase() === 'inprogress') {
+                state.taskStatus.inProgress = (
+                    parseInt(state.taskStatus?.inProgress as string) + 1
+                ).toString();
+            } else if (action.payload?.status?.toLowerCase() === 'Completed') {
+                state.taskStatus.inProgress = (
+                    parseInt(state.taskStatus?.completed as string) + 1
+                ).toString();
+            }
+
+            state.tasks.unshift(action.payload);
+        },
+
+        updateTaskInfo: (state, action) => {
+            const foundIndex: number = state.tasks.findIndex(
+                (item) => item.id === action.payload.id,
+            );
+            if (foundIndex > -1) {
+                if (action.payload?.status?.toLowerCase() === 'todo') {
+                    state.taskStatus.todo = (
+                        parseInt(state.taskStatus?.todo as string) + 1
+                    ).toString();
+                } else if (
+                    action.payload?.status?.toLowerCase() === 'inprogress'
+                ) {
+                    state.taskStatus.inProgress = (
+                        parseInt(state.taskStatus?.inProgress as string) + 1
+                    ).toString();
```

Comment:
Kind of repetitive logic we are using for both functions , can we move this logic to utility function

Reply:
Yea almost, we 'll refactor this in the new feature.